### PR TITLE
Dkg v2 improved

### PR DESCRIPTION
**Pull request by jbsv**

### CODE REVIEW

File: dkg/pedersen/handler.go [hunk 1]
```diff
@@ -28,13 +28,18 @@ import (
...
-// const chanSize = 10000
-
 // constant used in the logs
 const newState = "new state"
+const badState = "bad state: %v"
+const failedState = "failed to switch state: %v"
+
+// enumeration of the node type
+const (
+	unknownNode = iota
```
Comment (nkcr) [inline dkg/pedersen/handler.go:38]: you can only declare `iota` once, expressions are implicitly repeated in a paren­thesized const declaration

File: dkg/pedersen/handler.go [hunk 2]
```diff
@@ -670,12 +675,12 @@ func (h *Handler) handleDeal(ctx context.Context, msg types.Deal,
 	return nil
 }
 
-func (h *Handler) finalizeReshare(ctx context.Context, isCommonNode bool, out mino.Sender, from mino.Address) error {
+func (h *Handler) finalizeReshare(ctx context.Context, nodeType int, out mino.Sender, from mino.Address) error {
```
Comment (nkcr) [inline dkg/pedersen/handler.go:678]: would be even better to declare a type:

```
type nodeType byte
const (
    unknownNode nodeType = iota
    oldNode
    commonNode
...
)
...
func (h *Handler) finalizeReshare(ctx context.Context, t nodeType, out mino.Sender, from mino.Address) error {
```

**Review by nkcr — COMMENTED** [review]
Looks good, just a minor comment

File: dkg/pedersen/handler.go (hunk 2 above)
Comment (jbsv) [reply dkg/pedersen/handler.go:678]: Didn't know, but agree. What a cool language. thanks

**Pull request merged**